### PR TITLE
Fix var / package collisions and extra parens

### PR DESCRIPTION
**Pull request by Martin2112**
These lint fixes were mostly applied automatically but should be safe.

### CODE REVIEW

File: asn1/asn1.go [hunk 1]
```diff
@@ -936,15 +936,15 @@ func parseField(v reflect.Value, bytes []byte, initOffset int, params fieldParam
 		err = err1
 		return
 	case timeType:
-		var time time.Time
+		var tm time.Time
 		var err1 error
```
Comment (RJPercival) [inline asn1/asn1.go]: I wonder why `err1` exists, when they do `err = err1` at the bottom of this block anyway?

File: fixchain/fixer.go [hunk 2]
```diff
@@ -107,7 +107,7 @@ func min(a, b int) int {
 	return b
 }
 
-// sort.Sort(data Interface) for chainSlice - uses data.Len, data.Less & data.Swap.
+// Len implements sort.Sort(data Interface) for chainSlice - uses data.Len, data.Less & data.Swap.
```
Comment (RJPercival) [inline fixchain/fixer.go]: This reads more like documentation for `chainSlice` - move it up to that instead?

File: fixchain/hash.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ type bag struct {
 	certs []*x509.Certificate
 }
 
-// sort.Sort(data Interface) for bag - uses data.Len, data.Less & data.Swap
+// Len implements sort.Sort(data Interface) for bag - uses data.Len, data.Less & data.Swap
```
Comment (RJPercival) [inline fixchain/hash.go]: This reads more like documentation for `bag` - move it up to that instead?

File: submission/distributor.go [hunk 4]
```diff
@@ -247,7 +247,7 @@ func (d *Distributor) addSomeChain(ctx context.Context, rawChain [][]byte, asPre
 		if isPrecert {
 			inputType = "pre-"
 		}
-		return nil, fmt.Errorf("For add-%schain method %scertificate expected, got %scertificate", methodType, methodType, inputType)
+		return nil, fmt.Errorf("add-%schain method %scertificate expected, got %scertificate", methodType, methodType, inputType)
```
Comment (RJPercival) [inline submission/distributor.go]:
```suggestion
		return nil, fmt.Errorf("add-%schain method expected %scertificate, got %scertificate", methodType, methodType, inputType)
```

File: submission/loglist_manager.go [hunk 5]
```diff
@@ -70,7 +70,7 @@ func (llm *LogListManager) GetTwoLatestLogLists() (*LogListData, *LogListData) {
 // RefreshLogList reads Log List one time and runs updates if necessary.
 func (llm *LogListManager) RefreshLogList(ctx context.Context) (*LogListData, error) {
 	if llm.llr == nil {
-		return nil, fmt.Errorf("Log list manager has no log-list watcher to refresh Log List")
+		return nil, fmt.Errorf("log list manager has no log-list watcher to refresh Log List")
```
Comment (RJPercival) [inline submission/loglist_manager.go]: There seems to have been a concerted effort in this file to capitalise "Log" as recommended by @taknira. However, in this case, could rephrase the error message to use type names and prefix "the".
```suggestion
		return nil, fmt.Errorf("the LogListManager has no LogListRefresher")
```

File: trillian/ctfe/handlers.go [hunk 6]
```diff
@@ -146,7 +146,7 @@ type AppHandler struct {
 // ServeHTTP for an AppHandler invokes the underlying handler function but
 // does additional common error and stats processing.
 func (a AppHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	var status int
+	var st int
```
Comment (RJPercival) [inline trillian/ctfe/handlers.go]: I think this loses some clarity, given that the type name doesn't really indicate what this variable is. Perhaps name it "statusCode" instead?

File: asn1/asn1.go (hunk 1 above)
Comment (Martin2112) [reply asn1/asn1.go]: Reverted. The err1 thing isn't immediately obvious, SEP field on though.

File: fixchain/fixer.go (hunk 2 above)
Comment (Martin2112) [reply fixchain/fixer.go]: Done.

File: fixchain/hash.go (hunk 3 above)
Comment (Martin2112) [reply fixchain/hash.go]: Done.

File: trillian/ctfe/handlers.go (hunk 6 above)
Comment (Martin2112) [reply trillian/ctfe/handlers.go]: Done.

**Comment by Martin2112** [comment]
Looks like some of the stuff I reverted has somehow come back. I'll have another go tomorrow.

**Comment by Martin2112** [comment]
I think this is ready to merge now if Travis passes.

File: asn1/asn1.go (hunk 1 above)
Comment (RJPercival) [reply asn1/asn1.go]: Why did you revert this change?

**Comment by Martin2112** [comment]
I thought it was part of the code that would need upstream changes.

Martin

**Comment by RJPercival** [comment]
Despite your best efforts, this still ended up making changes to the `x509` and `asn1` directories.

**Comment by Martin2112** [comment]
Sorry, had a bit of a nightmare with this. I'll rewind them in a follow up.
They shouldn't change any functionality in the meantime.

Martin

**Pull request merged**